src/components/TopicSelector.jsx

```diff
@@ -1,50 +1,53 @@
 import { useEffect, useState } from "react";
 import { getTopics } from "../api";
-import { useNavigate } from "react-router-dom";
+import { useSearchParams } from "react-router-dom";
 
 export default function TopicSelector() {
 
-    const navigate = useNavigate();
-    const [topic, setTopic] = useState();
-    const { topics, isLoading, error } = loadTopics();
+    const [ searchParams, setSearchParams ] = useSearchParams();
+    const { topics, isLoading, error } = useLoadTopics();
+    const currentTopic = searchParams.get('topic') || '';
 
     const onChange = (event) => {
         event.preventDefault();
-        const selectedTopic = event.target.value;
-        setTopic(selectedTopic);
-        if(selectedTopic !== undefined && selectedTopic !== null && selectedTopic !== "All") {
-            navigate(`?topic=${selectedTopic}`);
-        } else {
-            navigate(``);
-        }
+        setSearchParams((params) => {
+            params.set('topic', event.target.value);
+            return params;
+        })
     }
 
+    if (isLoading) return <div>Loading...</div>;
+    if (error) return <div>Error: {`${error.code} ${error.message}`}</div>;
+
     return (
-        <>
-        <label htmlFor="topics">Category:</label>
-        <select name="topics" id="topics" onChange={onChange}>
-            {topics.map((topic)=> {
-                return topic === null ? <option key={"All"}>{"All"}</option> :
-                <option key={topic.slug} value={topic.slug}>{topic.slug}</option>
-            })}
-        </select> 
-        </>
+        <div id='topic-selector'>
+            <label htmlFor='topics'>Category:</label>
+            <select name='topics' id='topics' value={currentTopic} onChange={onChange}>
+                <option key={'All'}>{'All'}</option>
+                {topics.map((topic)=> {
+                    return <option key={topic.slug} value={topic.slug}>{topic.display}</option>
+                })}
+            </select> 
+        </div>
     )
 }
 
-function loadTopics() {
+function useLoadTopics() {
 
     const [ topics, setTopics ] = useState([]);
     const [isLoading, setLoading] = useState(true);
-    const [error, setError] = useState();
+    const [error, setError] = useState(null);
 
     useEffect(() => {
-
         const loadTopics = async () => {
+            setLoading(true);
             try {
-                const topics = await getTopics();
-                topics.unshift(null);
-                setTopics(topics);
+                const loadedTopics = await getTopics();
+                setTopics(loadedTopics.map((topic) => {
+                    topic['display'] = topic.slug.replace(/(^\w|\s\w)/g, m => m.toUpperCase()); // Capitalise first letter for display. Thank you stackexchange. 
```

Comment:
Thank you stackexchange, but also thank you Ben in advance for removing comments before pushing your code 😄